homeassistant/components/nextcloud/__init__.py

```diff
@@ -91,61 +92,30 @@
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Nextcloud integration."""
-    # Fetch Nextcloud Monitor api data
     conf = config[DOMAIN]
 
     try:
-        ncm = NextcloudMonitor(conf[CONF_URL], conf[CONF_USERNAME], conf[CONF_PASSWORD])
+        ncm = await hass.async_add_executor_job(
+            NextcloudMonitor, conf[CONF_URL], conf[CONF_USERNAME], conf[CONF_PASSWORD]
+        )
     except NextcloudMonitorError:
         _LOGGER.error("Nextcloud setup failed - Check configuration")
         return False
 
-    hass.data[DOMAIN] = get_data_points(ncm.data)
-    hass.data[DOMAIN]["instance"] = conf[CONF_URL]
-
-    def nextcloud_update(event_time):
-        """Update data from nextcloud api."""
-        try:
-            ncm.update()
-        except NextcloudMonitorError:
-            _LOGGER.error("Nextcloud update failed")
-            return False
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][conf[CONF_URL]] = coordinator = NextcloudDataUpdateCoordinator(
+        hass,
+        ncm,
+        conf,
+    )
```

Comment:
this level will be added with implementing config flow, because with config entries, it will be possible to add multiple instances of the nextcloud integration

Reply:
Yes, but later is later... and I think the recommendation will then be to use `hass.data[DOMAIN][config_entry.entry_id]` to differentiate.